src/OrchardCore/OrchardCore.Abstractions/Shell/IShellContextReleaseService.cs

```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.Environment.Shell;
+
+public interface IShellContextReleaseService
```

Comment:
I don't have a good suggestion, but a name that reflects that this is about queuing a release request until the end of the scope (not HTTP request) would be better. After all, if you just want to release the shell, why wouldn't you use `IShellHost`?

Reply:
if there is no good suggestions, the current name is good for me. But, if you can suggest a better name, we can look at it.

Reply:
DeferredShellContextReleaseService?

Reply:
@Piedone  Once I get response from you on the other items, I'll rename it.

Reply:
I renamed it to `IDeferredShellContextReleaseService` and `DeferredShellContextReleaseService` let's hope no one complaint about the length of the name :)